### PR TITLE
support network prune

### PR DESCRIPTION
**Pull request by junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

### CODE REVIEW

**Comment by junnplus** [comment]
```
C:\Windows\TEMP\cirrus-ci-build>call powershell hack/configure-windows-ci.ps1 
Installing the following packages:
git;golang
By installing, you accept licenses for the packages.
git not installed. An error occurred during installation:
 The remote server returned an error: (524).
git package files install failed with exit code 1. Performing other installation steps.
The install of git was NOT successful.
git not installed. An error occurred during installation:
 The remote server returned an error: (524).
golang not installed. An error occurred during installation:
 The remote server returned an error: (524).
golang package files install failed with exit code 1. Performing other installation steps.
The install of golang was NOT successful.
golang not installed. An error occurred during installation:
 The remote server returned an error: (524).
Chocolatey installed 0/2 packages. 2 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
Failures
 - golang (exited 1) - golang not installed. An error occurred during installation:
 The remote server returned an error: (524).
 - git (exited 1) - git not installed. An error occurred during installation:
 The remote server returned an error: (524).
```

File: README.md [hunk 1]
```diff
@@ -1034,6 +1035,16 @@ Remove one or more networks
 
 Usage: `nerdctl network rm NETWORK [NETWORK...]`
 
+### :whale: nerdctl network prune
+Remove all unused networks
+
+Usage: `nerdctl volume prune [OPTIONS]`
```
Comment (manugupt1) [inline README.md:1041]: should be network prune right?

File: cmd/nerdctl/network_prune_linux_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	base.Cmd("network", "create", testNetwork).AssertOK()
+	defer base.Cmd("network", "prune", "-f").Run()
+
+	tID := testutil.Identifier(t)
+	base.Cmd("run", "-d", "--net", testNetwork, "--name", tID, testutil.NginxAlpineImage).AssertOK()
+	defer base.Cmd("rm", "-f", tID).Run()
+
+	base.Cmd("network", "prune", "-f").AssertNoOut(testNetwork)
+	base.Cmd("stop", tID).AssertOK()
+	base.Cmd("network", "prune", "-f").AssertOutContains(testNetwork)
```
Comment (manugupt1) [inline cmd/nerdctl/network_prune_linux_test.go:37]: Can we please not use force here as force should not be the happy path?
We already have defer with force while creating network.

File: README.md (hunk 1 above)
Comment (junnplus) [reply README.md]: Oops

File: cmd/nerdctl/network_prune_linux_test.go (hunk 2 above)
Comment (junnplus) [reply cmd/nerdctl/network_prune_linux_test.go:37]: `force` no means force remove, it is for skip prompt.

File: cmd/nerdctl/network_prune.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+		removedNetworks = append(removedNetworks, net.Name)
+	}
+
+	if len(removedNetworks) > 0 {
+		fmt.Fprintln(cmd.OutOrStdout(), "Deleted Networks:")
+		for _, name := range removedNetworks {
+			fmt.Fprintln(cmd.OutOrStdout(), name)
+		}
+	}
+	return nil
```
Comment (AkihiroSuda) [inline cmd/nerdctl/network_prune.go:104]: Should we return err on failure?

Comment (junnplus) [reply cmd/nerdctl/network_prune.go:104]: Do you mean the error of removing the network? If so, I think logging is enough.

File: cmd/nerdctl/volume_rm.go [hunk 4]
```diff
@@ -45,21 +43,12 @@ func newVolumeRmCommand() *cobra.Command {
...
 func volumeRmAction(cmd *cobra.Command, args []string) error {
-	client, ctx, cancel, err := newClient(cmd)
-	if err != nil {
-		return err
-	}
-	defer cancel()
-	containers, err := client.Containers(ctx)
-	if err != nil {
-		return err
-	}
```
Comment (fahedouch) [inline cmd/nerdctl/volume_rm.go:56]: Why this refacto ? We may use these variables `client, ctx, cancel, err` in a futur improvement. It is better to keep them at this place

File: cmd/nerdctl/network_prune.go [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+		if err != nil {
+			return nil, err
+		}
+		status, err := task.Status(ctx)
+		if err != nil {
+			return nil, err
+		}
+		switch status.Status {
+		case containerd.Paused, containerd.Running:
+		default:
```
[the remark this answers is not in the document]
Comment (junnplus) [reply cmd/nerdctl/network_prune.go:130]: Not ignore

File: cmd/nerdctl/volume_rm.go (hunk 4 above)
Comment (junnplus) [reply cmd/nerdctl/volume_rm.go]: I remove it first, off topic

File: cmd/nerdctl/network_prune.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (junnplus) [reply cmd/nerdctl/network_prune.go:134]: Seems to be a problem, Docker is by checking the endpoint of the network

[the remark this answers is not in the document]
Comment (fahedouch) [reply cmd/nerdctl/network_prune.go:134]: Didn't get your point !

[the remark this answers is not in the document]
Comment (junnplus) [reply cmd/nerdctl/network_prune.go:134]: Docker doesn't clear the network in use, so the network of the running container is ignored here, but it has some problems.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**